controllers/discovery/searchUser.js

```diff
@@ -18,14 +18,27 @@ const SearchUser = async (req, res) => {
     });
 
   try {
+    let user_topics = await sequelize.query(
+      `select tp.topic_id from topics as tp
+      left join user_topics as utp on tp.topic_id = utp.topic_id
+      where utp.user_id = :userId`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          userId
+        }
+      }
+    );
+    let topicIds = user_topics.map((topic) => topic.topic_id);
+
     const users = await sequelize.query(
       `SELECT 
                 "User".*,
                 count("follower"."user_id_follower") 
                     AS "followersCount",
                 ARRAY( select name from topics as tp
                   left join user_topics as utp on tp.topic_id = utp.topic_id
-                  where utp.user_id = "User".user_id limit 3
+                  where utp.user_id = "User".user_id and tp.topic_id in (:topicIds) limit 3
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The second query that fetches users and their community info also has a potential SQL injection vulnerability with `topicIds`. This array of IDs is directly inserted into the SQL query without any sanitization or parameterization. This could allow an attacker to manipulate the SQL query by injecting malicious SQL code. To fix this, you should use parameterized queries or prepared statements to ensure that these values are properly escaped.
```diff
- where utp.user_id = "User".user_id and tp.topic_id in (:topicIds) limit 3
+ where utp.user_id = "User".user_id and tp.topic_id in (?) limit 3
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->